src/components/DataRoom/Hero/SafeLogo.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useRef, useEffect } from 'react'
+import { useFrame, useLoader } from '@react-three/fiber'
+import { GLTFLoader } from 'three/examples/jsm/loaders/GLTFLoader'
+import type { Group, Mesh } from 'three'
+import type { MotionValue } from 'framer-motion'
+
+type SafeLogoProps = {
+  yPosition: MotionValue<number>
+  offset: number
+}
+
+const DEFAULT_ROTATION = 0.005
+
+export function SafeLogo({ offset, yPosition }: SafeLogoProps) {
+  const groupRef = useRef<Group>(null)
+  const meshRef = useRef<Mesh>(null)
+  const gltf = useLoader(GLTFLoader, '/Safe_Logo.glb')
+
+  // Center the mesh along the z-axis based on its bounding box
+  useEffect(() => {
+    if (!meshRef.current) {
+      return
+    }
+    if (meshRef.current) {
+      meshRef.current.geometry.computeBoundingBox()
+      const boundingBox = meshRef.current.geometry.boundingBox
+      if (boundingBox) {
+        const zSize = boundingBox.max.z - boundingBox.min.z
+        meshRef.current.position.z = zSize / 2
+      }
+    }
```

Comment:
With the early return, we don't need to check that `meshRef.current` is defined:

```suggestion
    meshRef.current.geometry.computeBoundingBox()
    const boundingBox = meshRef.current.geometry.boundingBox
    if (boundingBox) {
      const zSize = boundingBox.max.z - boundingBox.min.z
      meshRef.current.position.z = zSize / 2
    }
```